2016-05-04  Yang Zhao  <yangzh@lnxsrv01.seas.ucla.edu>

* BUGFIX.H: Changed the timespec_cmp program so it does not rely
on subtraction.
		
--- timespec.h	2016-05-04 21:13:13.254571000 -0700
+++ BUGFIX.H	2016-05-04 20:59:15.731740000 -0700
@@ -45,8 +45,12 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  if (a.tv_sec == b.tv_sec)
+    return 0;
+  else if (a.tv_sec < b.tv_sec)
+    return -1;
+  else
+    return 1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
